modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -66,6 +64,20 @@ public void beforeAll(ExtensionContext context) {
         signalBeforeTestToContainers(lifecycleAwareContainers, testDescriptionFrom(context));
     }
 
+    private void startSharedContainers(
+        List<StoreAdapter> sharedContainersStoreAdapters,
+        Store store,
+        ExtensionContext context
+    ) {
+        Stream<StoreAdapter> storeAdapterStream = sharedContainersStoreAdapters.stream();
+        if (isParallelExecutionEnabled(context)) {
+            storeAdapterStream = storeAdapterStream.parallel();
```

Comment:
I think it would be better to use `Startables`

```java
Startables.deepStart(sharedContainersStoreAdapters.stream().map(adapter -> adapter.container))
            .join();
```